crates/re_query/src/archetype_view.rs

```diff
@@ -348,7 +348,7 @@ impl<A: Archetype> ArchetypeView<A> {
         components: impl IntoIterator<Item = ComponentWithInstances>,
     ) -> Self {
         Self {
-            row_id,
+            primary_row_id: row_id,
```

Comment:
The correct caching behavior of Tensor depends on this id being the id of the TensorData component. This will currently be the case for our definitions of Image / Tensor but may not always be the case if we introduce an archetype with an optional tensor.

I suspect this means we should do the book-keeping through so that every ComponentWithInstances knows its own RowId.

Since it's not a blocker today we can probably get away with just an issue / comment.

Reply:
Yes! I actually meant to add a comment about that and forgot. Will do.